### PR TITLE
Fix use of klog/v2 to use logr.Logger struct instead of interface

### PR DESCRIPTION
**Pull request by jacob-delgado**

### CODE REVIEW

**Comment by jacob-delgado** [comment]
I still need to do some testing to see that this works as intended. How do I test this @howardjohn / @ericvn ?

**Comment by howardjohn** [comment]
you can pull it into Istio/istio and then add --vklog 6 to istioctl and it
should log in our format. We do have pretty decent test coverage here as
well

File: log/logr.go [hunk 1]
```diff
@@ -76,27 +77,27 @@ func (zl *zapLogger) Error(err error, msg string, keysAndVals ...interface{}) {
 	}
 }
 
-func (zl *zapLogger) V(level int) logr.Logger {
-	return &zapLogger{
-		lvl:    zl.lvl + level,
-		l:      zl.l,
-		lvlSet: true,
+func (zl *zapLogger) V(int) logr.Logger {
```
Comment (howardjohn) [inline log/logr.go:80]: I am a bit suspicious that we don't use the level here? How does that work?

**Review by howardjohn — APPROVED** [review]
Seems to work but I don't actually understand how the V() stuff works :-)

**Pull request merged**